golang/pghoard_postgres_command_go.go

```diff
@@ -120,6 +120,16 @@ func restore_command(url string, output string) (int, error) {
 			}
 			output_path = path.Join(cwd, output)
 		}
+		// if file "<xlog>.pghoard.prefetch" exists, just move it to destination
+		xlogPrefetchPath := path.Join(path.Dir(output_path), xlog+".pghoard.prefetch")
+		_, err = os.Stat(xlogPrefetchPath)
```

Comment:
The `output_path` here is a relative path, such as `pg_wal/RECOVERYXLOG` (relative to the PG server, PG doc: `The path name is relative to the current working directory, i.e., the cluster's data directory.`), and PgHoard's working directory is under `/var/lib/pghoard`  on the server. This means the file existence checking here would not find the file whether exists.

Reply:
Relative path resolves to absolute path in same func higher (couple lines above).
Pghoard stores prefetch file in the same directory, so I think it's a good constraint, that we take prefetch file only from the same directory.
```
[root@egor-voynov-test-n1002 pg_wal]# ls -la
total 131084
drwx------.  3 postgres postgres          4096 Apr  2 10:04 .
drwx------. 19 postgres postgres          4096 Apr  2 09:51 ..
-rw-------.  1 postgres postgres      16777216 Apr  2 09:51 000000010000000000000002
-rw-------.  1 postgres postgres      16777216 Apr  2 09:51 000000010000000000000004
-rw-------.  1 postgres aiven_pghoard 16777216 Apr  2 10:04 000000010000000000000006.pghoard.prefetch
-rw-------.  1 postgres aiven_pghoard 16777216 Apr  2 10:04 000000010000000000000007.pghoard.prefetch
-rw-------.  1 postgres postgres      16777216 Apr  2 10:04 000000010000000000000008
-rw-------.  1 postgres aiven_pghoard 16777216 Apr  2 10:04 111
drwx------.  2 postgres postgres          4096 Apr  2 09:50 archive_status
-rw-------.  1 postgres postgres      16777216 Apr  2 09:56 test
-rw-------.  1 postgres postgres      16777216 Apr  2 09:51 test_
[root@egor-voynov-test-n1002 pg_wal]# pwd
/srv/aiven-persistent/1002-service/pghome/15/data/pg_wal
```

Reply:
> The `output_path` here is a relative path, such as `pg_wal/RECOVERYXLOG` (relative to the PG server, PG doc: `The path name is relative to the current working directory, i.e., the cluster's data directory.`), and PgHoard's working directory is under `/var/lib/pghoard` on the server. This means the file existence checking here would not find the file whether exists. I think we have to introduce the data directory as input argument of the command.

Sorry, I am mistaken. I thought `restore_command` was executed by PgHoard. But if it is executed by PG itself, the path should not have a problem. 

Disregard my first comment.

Reply:
But there is output_path wrapped by `path.Dir` function, which returns directory:
`path.Join(path.Dir(output_path), xlog+".pghoard.prefetch")`
Maybe I should split it to 2 lines like this:
```
output_path_dir = path.Dir(output_path)
path.Join(output_path_dir, xlog+".pghoard.prefetch")
```
Following your example:
```
output_path="/var/lib/pgsql/15/data/pg_wal/RECOVERYXLOG"
output_path_dir="/var/lib/pgsql/15/data/pg_wal"
xlog="000000010000000000000006"
xlogPrefetchPath="/var/lib/pgsql/15/data/pg_wal/000000010000000000000006.pghoard.prefetch"
```

Reply:
Ok, I was blind again, then looks good to me.